src/shared/componentutils/utils.go

```diff
@@ -1,13 +1,54 @@
 package componentutils
 
 import (
+	"errors"
+	"fmt"
 	"github.com/otterize/intents-operator/src/shared/otterizecloud/otterizecloudclient"
 	"github.com/otterize/intents-operator/src/shared/telemetries/componentinfo"
+	sharedconfig "github.com/otterize/network-mapper/src/shared/config"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
+	"os"
+	"time"
 )
 
 func SetCloudClientId() {
 	// Cloud client ID is set inside the cloud client initialization, this method is meant to be used from components
 	// that don't use the cloud client, but still need it for their telemetry data.
 	componentinfo.SetGlobalCloudClientId(viper.GetString(otterizecloudclient.ApiClientIdKey))
 }
+
+func WaitAndSetContextId() error {
```

Comment:
this should receive a context as an argument and select on it, to avoid blocking forever and ignoring shutdown triggers.